src/options.js

```diff
@@ -0,0 +1,215 @@
+var allPermissions = {}
+var extOptions     = {}
+
+function getAuthorizedSites() {
+    chrome.permissions.getAll( function (permissions) {
+        allPermissions = permissions;
+    });
+}
+
+function showAuthorizedSites() {
+    chrome.permissions.getAll( function (permissions) {
+        allPermissions = permissions;
+        chrome.storage.sync.get({
+            public_gist_enabled: true,
+            public_github_enabled: true
+        },
+        function(options) {
+            extOptions = options;
+            $.each(permissions.origins, function (key, value) {
+                if (value == "https://gist.github.com/*" || value == "https://github.com/*") {
```

Comment:
It could be but I'm not sure it would be much of an improvement. These two are the default sites/permissions in the manifest and will always be formatted as such (basically the default functionality of the extension). This bit of code is just making sure that if we are drawing the options page buttons for the default functionality we provide the option to turn on/off respectively instead of completely remove.